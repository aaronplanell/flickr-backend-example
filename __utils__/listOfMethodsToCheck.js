export const listOfMethods = [
    'flickr.activity.userPhotos',
    'flickr.auth.oauth.checkToken',
    'flickr.auth.checkToken',
    'flickr.auth.getFullToken',
    'flickr.cameras.getBrandModels',
    'flickr.activity.userComments',
    'flickr.blogs.getServices',
    'flickr.auth.getToken',
    'flickr.blogs.getList',
    'flickr.collections.getInfo',
    'flickr.cameras.getBrands',
    'flickr.collections.getTree',
    'flickr.auth.oauth.getAccessToken',
    'flickr.commons.getInstitutions',
    'flickr.contacts.getList',
    'flickr.contacts.getListRecentlyUploaded',
    'flickr.auth.getFrob',
    'flickr.contacts.getPublicList',
    'flickr.contacts.getTaggingSuggestions',
    'flickr.favorites.getList',
    'flickr.favorites.getContext',
    'flickr.favorites.getPublicList',
    'flickr.galleries.getInfo',
    'flickr.galleries.getPhotos',
    'flickr.galleries.getListForPhoto',
    'flickr.groups.browse',
    'flickr.groups.discuss.replies.getInfo',
    'flickr.galleries.getList',
    'flickr.groups.discuss.replies.getList',
    'flickr.groups.discuss.topics.getInfo',
    'flickr.groups.discuss.topics.getList',
    'flickr.groups.getInfo',
    'flickr.groups.pools.getContext',
    'flickr.groups.members.getList',
    'flickr.groups.pools.getPhotos',
    'flickr.groups.search',
    'flickr.interestingness.getList',
    'flickr.groups.pools.getGroups',
    'flickr.machinetags.getPredicates',
    'flickr.machinetags.getNamespaces',
    'flickr.machinetags.getRecentValues',
    'flickr.machinetags.getPairs',
    'flickr.machinetags.getValues',
    'flickr.panda.getList',
    'flickr.panda.getPhotos',
    'flickr.people.findByEmail',
    'flickr.people.getLimits',
    'flickr.people.findByUsername',
    'flickr.people.getInfo',
    'flickr.people.getGroups',
    'flickr.people.getPhotos',
    'flickr.people.getPhotosOf',
    'flickr.people.getPublicGroups',
    'flickr.people.getPublicPhotos',
    'flickr.photos.comments.getList',
    'flickr.photos.comments.getRecentForContacts',
    'flickr.people.getUploadStatus',
    'flickr.photos.geo.getLocation',
    'flickr.photos.geo.getPerms',
    'flickr.photos.geo.photosForLocation',
    'flickr.photos.getAllContexts',
    'flickr.photos.getContactsPhotos',
    'flickr.photos.getContext',
    'flickr.photos.getExif',
    'flickr.photos.getCounts',
    'flickr.photos.getInfo',
    'flickr.photos.getNotInSet',
    'flickr.photos.getPerms',
    'flickr.photos.getFavorites',
    'flickr.photos.getRecent',
    'flickr.photos.getPopular',
    'flickr.photos.getWithGeoData',
    'flickr.photos.getContactsPublicPhotos',
    'flickr.photos.getUntagged',
    'flickr.photos.getWithoutGeoData',
    'flickr.photos.getSizes',
    'flickr.photos.licenses.getInfo',
    'flickr.photos.people.getList',
    'flickr.photos.recentlyUpdated',
    'flickr.photos.suggestions.getList',
    'flickr.photos.search',
    'flickr.photos.upload.checkTickets',
    'flickr.photosets.comments.getList',
    'flickr.photosets.getContext',
    'flickr.photosets.getList',
    'flickr.photosets.getPhotos',
    'flickr.photosets.getInfo',
    'flickr.places.find',
    'flickr.places.findByLatLon',
    'flickr.places.getChildrenWithPhotosPublic',
    'flickr.places.getInfo',
    'flickr.places.getInfoByUrl',
    'flickr.places.getShapeHistory',
    'flickr.places.getTopPlacesList',
    'flickr.places.placesForBoundingBox',
    'flickr.places.getPlaceTypes',
    'flickr.places.placesForTags',
    'flickr.places.placesForUser',
    'flickr.places.resolvePlaceId',
    'flickr.prefs.getContentType',
    'flickr.places.tagsForPlace',
    'flickr.places.resolvePlaceURL',
    'flickr.prefs.getGeoPerms',
    'flickr.prefs.getHidden',
    'flickr.places.placesForContacts',
    'flickr.prefs.getPrivacy',
    'flickr.profile.getProfile',
    'flickr.prefs.getSafetyLevel',
    'flickr.push.getSubscriptions',
    'flickr.push.subscribe',
    'flickr.push.getTopics',
    'flickr.stats.getCollectionReferrers',
    'flickr.stats.getCollectionDomains',
    'flickr.stats.getCollectionStats',
    'flickr.stats.getCSVFiles',
    'flickr.stats.getPhotoReferrers',
    'flickr.stats.getPhotoDomains',
    'flickr.push.unsubscribe',
    'flickr.stats.getPhotosetReferrers',
    'flickr.reflection.getMethods',
    'flickr.stats.getPhotosetDomains',
    'flickr.stats.getPhotosetStats',
    'flickr.stats.getPhotoStats',
    'flickr.stats.getPhotostreamDomains',
    'flickr.reflection.getMethodInfo',
    'flickr.stats.getPhotostreamReferrers',
    'flickr.stats.getPopularPhotos',
    'flickr.tags.getClusterPhotos',
    'flickr.tags.getClusters',
    'flickr.stats.getTotalViews',
    'flickr.tags.getListPhoto',
    'flickr.tags.getHotList',
    'flickr.stats.getPhotostreamStats',
    'flickr.tags.getListUser',
    'flickr.tags.getListUserPopular',
    'flickr.tags.getMostFrequentlyUsed',
    'flickr.test.echo',
    'flickr.tags.getRelated',
    'flickr.test.null',
    'flickr.tags.getListUserRaw',
    'flickr.test.login',
    'flickr.urls.lookupGallery',
    'flickr.urls.getUserPhotos',
    'flickr.urls.getGroup',
    'flickr.urls.getUserProfile',
    'flickr.urls.lookupGroup',
    'flickr.urls.lookupUser'
];
