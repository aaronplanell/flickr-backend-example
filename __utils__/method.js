import request from './request';
import { APPLICATION_PORT } from '../src/constants';

export const callMethod = (methodName) => {
  return request('http://127.0.0.1:' +  APPLICATION_PORT + '/' + methodName).then(resultAsJson => JSON.parse(resultAsJson));
};
