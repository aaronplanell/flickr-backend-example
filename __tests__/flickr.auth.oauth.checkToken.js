import { callMethod } from '../__utils__/method';

it('flickr.auth.oauth.checkToken', () => {
  return callMethod('flickr.auth.oauth.checkToken')
    .then(result => {
      expect(result.stat).toEqual('ok');
      expect(result).toMatchSnapshot();
    });
});
