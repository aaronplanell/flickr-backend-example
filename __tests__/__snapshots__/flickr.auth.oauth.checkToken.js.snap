// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flickr.auth.oauth.checkToken 1`] = `
Object {
  "oauth": Object {
    "perms": Object {
      "_content": "read",
    },
    "token": Object {
      "_content": "72157679332129251-4c25dc1c269191d0",
    },
    "user": Object {
      "fullname": "Aaron Planell López",
      "nsid": "148575064@N08",
      "username": "Aaron Planell López",
    },
  },
  "stat": "ok",
}
`;
